src/applications/simple-forms/mock-simple-forms-patterns/arrayBuilder/components/ArrayBuilderChapter.jsx

```diff
@@ -0,0 +1,373 @@
+ *     path: '/employer',
+ *     uiSchema: ...,
+ *     schema: ...
+ *   }),
+ *   itemLastPage: pageBuilder.itemLastPage({
+ *     title: 'Name and address of employer or unit',
+ *     path: '/employer',
+ *     uiSchema: ...,
+ *     schema: ...
+ *   }),
+ * }
+ * ```
+ *
+ * @param {(pageBuilder: ArrayBuilderPages) => ArrayBuilderConfig} configFunction
+ * @returns {FormConfigChapter}
+ */
+export default function arrayBuilderChapter(configFunction) {
+  let itemFirstPagePath;
+  let summaryPath;
+  let hasItemsKey;
+
+  if (!configFunction || typeof configFunction !== 'function') {
+    throwErrorNoConfigFunction();
+  }
+
+  /**
+   * @type {{
+   *   [key: string]: (config: FormConfigPage) => FormConfigPage,
+   * }}
+   */
+  const pageBuilder = {
+    summaryPage: pageConfig => {
+      summaryPath = pageConfig.path;
+      try {
+        hasItemsKey = determineYesNoField(pageConfig.uiSchema);
+      } catch (e) {
+        throwMissingYesNoField();
+      }
+      if (!hasItemsKey) {
+        throwMissingYesNoField();
+      }
+    },
+    itemFirstPage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+      itemFirstPagePath = getBasePath(pageConfig.path);
+    },
+    itemMiddlePage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+    },
+    itemLastPage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+    },
+    itemSinglePage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+    },
+  };
+
+  // run configFunction once to get initial options
+  const { options } = configFunction(pageBuilder);
+  if (!options) {
+    throwErrorNoOptions();
+  }
+
+  const requiredChapterOpts = ['arrayPath', 'noun', 'nextChapterPath'];
+  verifyRequiredOptions('chapter', requiredChapterOpts, options);
+
+  const {
+    noun,
+    arrayPath,
+    nextChapterPath,
+    reviewPath = '/review-and-submit',
+    isIncompleteItem = item => !item?.name, // TODO: update this to use user param
+  } = options;
+
+  const nounSingular = noun;
+  const nounPlural = pluralize(noun);
+
+  const CustomPageItem = ArrayBuilderItemPage({
+    arrayPath,
+    nounSingular,
+    nounPlural,
+    summaryRoute: summaryPath,
+  });
+
+  const SummaryCards = (
+    <ArrayBuilderCards
+      cardDescription={itemData =>
+        `${itemData?.dateStart} - ${itemData?.dateEnd}`
+      }
+      arrayPath={arrayPath}
+      nounSingular={nounSingular}
+      nounPlural={nounPlural}
+      isIncomplete={isIncompleteItem}
+      editItemBasePathUrl={itemFirstPagePath}
+    />
+  );
+
+  /** @returns {FormConfigPage} */
+  const commonItemPageConfig = depends => ({
+    showPagePerItem: true,
+    allowPathWithNoItems: true,
+    arrayPath,
+    CustomPage: CustomPageItem,
+    CustomPageReview: () => null,
```

Comment:
What does this look like on the review & submit page? It's not customizable?

Reply:
you can add CustomPageReview, but we will default to just showing cards for the first iteration.